src/Raven.Server/Indexing/LuceneVoronDirectory.cs

```diff
@@ -160,6 +165,19 @@ public IDisposable SetTransaction(Transaction tx, out IState state)
 
         protected override void Dispose(bool disposing)
         {
+            // should be empty since we remove it from the dictionary on file dispose
+            Debug.Assert(_voronIndexFiles.Count == 0);
```

Comment:
Let's change this to a real exception maybe?

Reply:
Wouldn't it cause index dispose to throw?

Reply:
Lucene in general is copying back and forth the outputs, so you might be right, but if you are then Debug.Assert will kill the runtime when tests are executed in debug.

Reply:
done